app/models/manageiq/providers/google/refresh_helper_methods.rb

```diff
@@ -2,15 +2,20 @@ module ManageIQ::Providers::Google::RefreshHelperMethods
   extend ActiveSupport::Concern
 
   def process_collection(collection, key)
+    process_flatten_collection(collection, key) { |x| [yield(x)] }
+  end
+
+  def process_flatten_collection(collection, key)
```

Comment:
Agreed, it's more readable; only downside is that there is a bit more magic, but its minimal and useful so I like it.  Will fix it.